### ISSUE TITLE
Track number in the file name

### ISSUE THREAD

**Issue by bitmarker** [jrnewell/spotify-ripper#5]
Hello!

When saving a playlist the song order gets lost, because the files are ordered alphabetically. I would suggest prepend the index number to the file name.

Like this. In Line 292:

self.mp3_file = os.path.join(base_dir, ("%.3d" % idx) + " - " + artist + " - " + track_name + ".mp3").encode('ascii', 'ignore')

**Comment by jrnewell** [comment]
OK.  I added that option ```--Flat [-F]``` in commit bd603be.  I think if I get another format request I will add a config file option so people can write their own custom output format.

**Comment by atcapollo** [comment]
I've edited ./spotify-ripper/ripper.py to include the track number. But this is not getting picked up when executed, am I editing the wrong file?

**Comment by ellekz** [comment]
I'm running 1.2.5 on a Debian machine and this doesn't seem to work for me?  

My config.ini:
```
[main]
key = /Downloads/_SpotifyRipper/Config/spotify_appkey.key
ascii = True
bitrate = 320
cbr = VBR
vbr = 0
Flat = True
user = REDACTED

```

Run:
```
spotify-ripper -S /Downloads/_SpotifyRipper/Config -F -d /Downloads/_SpotifyRipper/Playlist_WorkoutMix spotify:user:REDACTED:playlist:REDACTED
```

All files end up as `/Downloads/_SpotifyRipper/Playlist_WorkoutMix/Artist - Song.mp3`. Am I missing something?

**Comment by jrnewell** [comment]
No, you aren't missing anything.  It looks like there is an issue with the fact that the config parser is case-insensitive when it parses an INI file so it is mis-interpreting your ```Flat = True``` as ```flat = True``` (and when both options are set it prefers the lower-case option, which doesn't include the index in the file name).

I don't really like the name I gave that option, so I renamed it to ``-F | --flat-with-index`` to make it more clear and to avoid this issue.  Upgrade to version ``1.3.0`` and change your config file to:

```
[main]
key = /Downloads/_SpotifyRipper/Config/spotify_appkey.key
ascii = True
bitrate = 320
cbr = VBR
vbr = 0
flat_with_index = True
user = REDACTED
```

and it should work now.  Let me know if you still have issues.
